contracts/batcher/TwoWayBatcher.sol

```diff
@@ -0,0 +1,63 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.13;
+
+import "@equilibria/root/number/types/UFixed18.sol";
+import "@equilibria/root/token/types/Token18.sol";
+import "@equilibria/root/token/types/Token6.sol";
+import "@equilibria/root/control/unstructured/UReentrancyGuard.sol";
+import "./Batcher.sol";
+
+contract TwoWayBatcher is UReentrancyGuard, Batcher {
+    using UFixed18Lib for UFixed18;
+    using Token18Lib for Token18;
+    using Token6Lib for Token6;
+
+    event USDCLoaned(address indexed depositor, UFixed18 amount);
+    event USDCRepaid(address indexed depositor, UFixed18 amount);
+
+    UFixed18 usdcLoansOutstanding;
+    mapping(address => UFixed18) depositorToUSDCLoanAmount;
+
+    constructor(IEmptySetReserve reserve, Token18 dsu, Token6 usdc)
+    Batcher(reserve, dsu, usdc)
+    { }
+
+    function loanUSDC(UFixed18 amount) external nonReentrant {
+        USDC.pull(msg.sender, amount, true);
+
+        usdcLoansOutstanding = usdcLoansOutstanding.add(amount);
+        depositorToUSDCLoanAmount[msg.sender] = depositorToUSDCLoanAmount[msg.sender].add(amount);
+
+        emit USDCLoaned(msg.sender, amount);
+    }
+
+    function repayUSDC(UFixed18 amount) external nonReentrant {
+        if (USDC.balanceOf(address(this)).lt(amount)) {
+            rebalance();
```

Comment:
I think we should just rebalance every time to get stuff into a good state.

Reply:
Actually I think we will still be in a good state after repayment. When you repay (or loan) you're increasing or decreasing the USDC balance by the same amount as  you change `usdcLoansOutstanding` (target USDC amount). So this will net out equally and you should be in a similarly good state after the repayment.

Reply:
If that's correct I think just hitting rebalance at the start of `loan` and `repay` would be nice as a cleanliness thing.